crates/nargo_cli/src/cli/init_cmd.rs

```diff
@@ -5,13 +5,22 @@ use super::{NargoConfig, CARGO_PKG_VERSION};
 use acvm::Backend;
 use clap::Args;
 use nargo::constants::{PKG_FILE, SRC_DIR};
+use nargo::package::PackageType;
 use std::path::PathBuf;
 
 /// Create a Noir project in the current directory.
 #[derive(Debug, Clone, Args)]
-pub(crate) struct InitCommand;
+pub(crate) struct InitCommand {
+    /// Use a library template
+    #[clap(long)]
+    pub(crate) lib: bool,
 
-const EXAMPLE: &str = r#"fn main(x : Field, y : pub Field) {
+    /// Use a binary template [default]
+    #[clap(long)]
+    pub(crate) bin: bool,
```

Comment:
Clap should have an option to make these mutually exclusive.

Reply:
Resolved by @kobyhallx's suggestion